compiler-plugin/src/main/java/io/ballerina/stdlib/websub/action/SubscriberServiceTemplateGenerationAction.java

```diff
@@ -0,0 +1,114 @@
+
+import static io.ballerina.stdlib.websub.action.CodeActionUtil.findNode;
+
+/**
+ * {@code EmptySubscriberServiceTemplateGenerateAction} generates empty `websub:SubscriberService` code-snippet.
+ */
+public class SubscriberServiceTemplateGenerationAction implements CodeAction {
+    private static final String NODE_LOCATION = "node.location";
+    private static final String CODE_ACTION_TITLE = "Add subscriber service";
+
+    private final Service websubServiceSnippet;
+
+    public SubscriberServiceTemplateGenerationAction() {
+        this.websubServiceSnippet = CodeActionUtil.constructSubscriberService();
+    }
+
+    @Override
+    public List<String> supportedDiagnosticCodes() {
+        return List.of(WebSubDiagnosticCodes.WEBSUB_202.getCode());
+    }
+
+    @Override
+    public Optional<CodeActionInfo> codeActionInfo(CodeActionContext context) {
+        return Optional.ofNullable(context.diagnostic())
+                .filter(d -> Objects.nonNull(d.location()))
+                .map(this::constructCodeActionInfo);
+    }
+
+    private CodeActionInfo constructCodeActionInfo(Diagnostic diagnostic) {
+        CodeActionArgument locationArg = CodeActionArgument.from(NODE_LOCATION, diagnostic.location().lineRange());
+        return CodeActionInfo.from(CODE_ACTION_TITLE, List.of(locationArg));
+    }
+
+    @Override
+    public List<DocumentEdit> execute(CodeActionExecutionContext executionContext) {
+        Optional<LineRange> lineRangeOpt = executionContext.arguments().stream()
+                .filter(arg -> NODE_LOCATION.equals(arg.key()))
+                .map(arg -> arg.valueAs(LineRange.class))
+                .findFirst();
+        if (lineRangeOpt.isEmpty()) {
+            return Collections.emptyList();
+        }
+        LineRange lineRange = lineRangeOpt.get();
+        SyntaxTree syntaxTree = executionContext.currentDocument().syntaxTree();
+        NonTerminalNode node = findNode(syntaxTree, lineRange);
+        if (!(node instanceof ServiceDeclarationNode)) {
+            return Collections.emptyList();
+        }
+        List<TextEdit> textEdits = retrieveRequiredTextEdits((ServiceDeclarationNode) node);
+        TextDocumentChange change = TextDocumentChange.from(textEdits.toArray(new TextEdit[0]));
+        return Collections.singletonList(
+                new DocumentEdit(executionContext.fileUri(), SyntaxTree.from(syntaxTree, change)));
+    }
+
+    private List<TextEdit> retrieveRequiredTextEdits(ServiceDeclarationNode serviceDeclarationNode) {
+        TextRange annotationTextRange = TextRange.from(
+                serviceDeclarationNode.serviceKeyword().textRange().startOffset(), 0);
+        TextEdit annotationsEdit = TextEdit.from(
+                annotationTextRange, String.format("%s%s", websubServiceSnippet.getAnnotationSnippet(), Constants.LS));
+        TextRange functionTextRange = TextRange.from(
+                serviceDeclarationNode.openBraceToken().textRange().endOffset(), 0);
```

Comment:
Giving the `0` as the length will result in unnecessary new lines at the completion. Better to take the gap in between the last member of the function and the close brace.